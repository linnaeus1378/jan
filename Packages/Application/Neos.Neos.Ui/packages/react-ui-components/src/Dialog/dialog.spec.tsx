import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Portal} from 'react-portal';

import DialogWithEscape, {DialogProps, DialogWithoutEscape} from './dialog';

describe('<Dialog/>', () => {
    const props: DialogProps = {
        actions: ['Foo 1', 'Foo 2'],
        children: 'Foo children',
        isOpen: true,
        autoFocus: true,
        onRequestClose: () => null,
        style: 'wide',
        type: 'error',
        theme: {
            'dialog': 'dialogClassName',
            'dialog--narrow': 'narrowClassName',
            'dialog--wide': 'wideClassName',
            'dialog--jumbo': 'jumboClassName',
            'dialog--success': 'successClassName',
            'dialog--warn': 'warnClassName',
            'dialog--error': 'errorClassName',
            'dialog__actions': 'actionsClassName',
            'dialog__body': 'bodyClassName',
            'dialog__closeBtn': 'closeBtnClassName',
            'dialog__contents': 'contentsClassName',
            'dialog__contentsPosition': 'contentsPositionClassName',
            'dialog__title': 'titleClassName',
        },
        title: 'Foo title',
    };

    it('Portal should render correctly.', () => {
        const wrapper = shallow(<DialogWithEscape {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Content should render correctly.', () => {
        const wrapper = shallow(<DialogWithoutEscape {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render the "dialog--wide" className from the "theme" prop if the style is wide.', () => {
        const wrapper = shallow(<DialogWithEscape {...props} style="wide"/>);
        const portal = wrapper.find(Portal);
        const section = portal.find('section');

        expect(section.prop('className')).toContain('wideClassName');
    });

    it('should render the "dialog--jumbo" className from the "theme" prop if the style is jumbo.', () => {
        const wrapper = shallow(<DialogWithEscape {...props} style="jumbo"/>);
        const portal = wrapper.find(Portal);
        const section = portal.find('section');

        expect(section.prop('className')).toContain('jumboClassName');
    });

    it('should render the "dialog--narrow" className from the "theme" prop if the style is narrow.', () => {
        const wrapper = shallow(<DialogWithEscape {...props} style="narrow"/>);
        const portal = wrapper.find(Portal);
        const section = portal.find('section');

        expect(section.prop('className')).toContain('narrowClassName');
    });

    it('should render the actions if passed.', () => {
        const wrapper = shallow(<DialogWithoutEscape {...props}/>);

        expect(wrapper.html().includes('Foo 1')).toBeTruthy();
        expect(wrapper.html().includes('Foo 2')).toBeTruthy();
    });
});
