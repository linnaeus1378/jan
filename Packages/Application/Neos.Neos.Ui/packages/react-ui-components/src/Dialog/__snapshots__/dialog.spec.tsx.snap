// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog/> Content should render correctly. 1`] = `
<div
  className="contentsPositionClassName"
  tabIndex={0}
>
  <div
    className="contentsClassName"
  >
    <div
      className="titleClassName"
    >
      Foo title
    </div>
    <div
      className="bodyClassName errorClassName dialog__body"
    >
      Foo children
    </div>
    <div
      className="actionsClassName"
    >
      <span
        key="0/.0"
      >
        Foo 1
      </span>
      <span
        key="1/.1"
      >
        Foo 2
      </span>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> Portal should render correctly. 1`] = `
<CloseOnEscape
  onEscape={[Function]}
>
  <Portal>
    <section
      autoFocus={true}
      className="dialogClassName wideClassName errorClassName"
      role="dialog"
      tabIndex={0}
    >
      <clickOutside(DialogWithoutEscape)
        actions={
          Array [
            "Foo 1",
            "Foo 2",
          ]
        }
        autoFocus={true}
        isOpen={true}
        onRequestClose={[Function]}
        style="wide"
        theme={
          Object {
            "dialog": "dialogClassName",
            "dialog--error": "errorClassName",
            "dialog--jumbo": "jumboClassName",
            "dialog--narrow": "narrowClassName",
            "dialog--success": "successClassName",
            "dialog--warn": "warnClassName",
            "dialog--wide": "wideClassName",
            "dialog__actions": "actionsClassName",
            "dialog__body": "bodyClassName",
            "dialog__closeBtn": "closeBtnClassName",
            "dialog__contents": "contentsClassName",
            "dialog__contentsPosition": "contentsPositionClassName",
            "dialog__title": "titleClassName",
          }
        }
        title="Foo title"
        type="error"
      >
        Foo children
      </clickOutside(DialogWithoutEscape)>
    </section>
  </Portal>
</CloseOnEscape>
`;
